sphinx_search/static/js/rtd_sphinx_search.js

```diff
@@ -550,28 +554,78 @@ const fetchAndGenerateResults = (api_endpoint, parameters, projectName) => {
  * This html structure will serve as the boilerplate
  * to show our search results.
  *
+ * @param {Array} filters: filters to be applied to the search.
+ *  {["Filter name", "Filter value"]}
  * @return {String} initial html structure
  */
-const generateAndReturnInitialHtml = () => {
-    let innerHTML =
-        '<div class="search__outer"> \
-            <div class="search__cross" title="Close"> \
-                <!--?xml version="1.0" encoding="UTF-8"?--> \
-                <svg class="search__cross__img" width="15px" height="15px" enable-background="new 0 0 612 612" version="1.1" viewBox="0 0 612 612" xml:space="preserve" xmlns="http://www.w3.org/2000/svg"> \
-                    <polygon points="612 36.004 576.52 0.603 306 270.61 35.478 0.603 0 36.004 270.52 306.01 0 576 35.478 611.4 306 341.41 576.52 611.4 612 576 341.46 306.01"></polygon> \
-                </svg> \
-            </div> \
-            <input class="search__outer__input" placeholder="Search ..."> \
-            <span class="bar"></span> \
-        </div> \
-        <div class="rtd__search__credits"> \
-            Search by <a href="https://readthedocs.org/">Read the Docs</a> & <a href="https://readthedocs-sphinx-search.readthedocs.io/en/latest/">readthedocs-sphinx-search</a> \
-        </div>';
+const generateAndReturnInitialHtml = (filters) => {
+    let innerHTML = `
+        <div class="search__outer">
+          <div class="search__cross" title="Close">
+            <!--?xml version="1.0" encoding="UTF-8"?-->
+            <svg class="search__cross__img" width="15px" height="15px" enable-background="new 0 0 612 612" version="1.1" viewBox="0 0 612 612" xml:space="preserve" xmlns="http://www.w3.org/2000/svg">
+              <polygon points="612 36.004 576.52 0.603 306 270.61 35.478 0.603 0 36.004 270.52 306.01 0 576 35.478 611.4 306 341.41 576.52 611.4 612 576 341.46 306.01"></polygon>
+            </svg>
+          </div>
+          <input class="search__outer__input" placeholder="Search...">
+          <div class="bar"></div>
+          <div class="search__filters">
+            <ul>
+            </ul>
+          </div>
+        </div>
+        <div class="rtd__search__credits">
+          Search by <a href="https://readthedocs.org/">Read the Docs</a> & <a href="https://readthedocs-sphinx-search.readthedocs.io/en/latest/">readthedocs-sphinx-search</a>
+        </div>
+    `;
 
     let div = createDomNode("div", {
         class: "search__outer__wrapper search__backdrop",
     });
     div.innerHTML = innerHTML;
+
+    let filters_list = div.querySelector(".search__filters ul");
+    const config = getConfig();
+    // Add filters below the search box if present.
+    if (filters.length > 0) {
+      let li = createDomNode("li", {"class": "search__filters__title"});
```

Comment:
I don't like too much the pattern of "generating nodes on the fly" a lot and I'd prefer to use literal strings if possible. This way is a lot easier to read and follow the code.

Reply:
If you mean building the strings manually, we need to be more cautious about user input, a better pattern can be just building the skeleton first, and then change that (https://github.com/readthedocs/readthedocs-sphinx-search/issues/117#issuecomment-1457196360)